### PR TITLE
Add logging for file shares

### LINKED ISSUE #2318: File share logging
part of #1922

### PR DESCRIPTION
**Pull request by JimMadge**
## :white_check_mark: Checklist

- [x] You have given your pull request a meaningful title (_e.g._ `Enable foobar integration` rather than `515 foobar`).
- [x] You are targeting the appropriate branch. If you're not certain which one this is, it should be **`develop`**.
- [x] Your branch is up-to-date with the **target branch** (it probably was when you started, but it may have changed since then).

### :vertical_traffic_light: Depends on

#2310 

### :closed_umbrella: Related issues

Closes #2318 
Contributes to #1922 

### :microscope: Tests

Tested on a fresh SRE deployment

Screenshot 2024-11-29 at 09 44 24

### CODE REVIEW

**Comment by jemrobinson** [comment]
What does this actually log? Is it file creation/deletion etc?

File: data_safe_haven/infrastructure/programs/declarative_sre.py [hunk 1]
```diff
@@ -198,20 +198,6 @@ def __call__(self) -> None:
...
-            SREMonitoringProps(
-                dns_private_zones=dns.private_zones,
-                location=self.config.azure.location,
-                resource_group_name=resource_group.name,
-                subnet=networking.subnet_monitoring,
-                timezone=self.config.sre.timezone,
-            ),
-            tags=self.tags,
-        )
-
```
Comment (jemrobinson) [inline data_safe_haven/infrastructure/programs/declarative_sre.py:214]: Why is this removed? Have we dropped/renamed/separated the SREMonitoringComponent somewhere else?

File: data_safe_haven/infrastructure/programs/sre/data.py [hunk 2]
```diff
@@ -425,6 +426,47 @@ def __init__(
...
+                        "enabled": False,
+                    },
+                }
+            ],
+            resource_uri=storage_account_data_configuration.id.apply(
+                # This is the URI of the fileService resource which is automatically
+                # created.
+                lambda resource_id: resource_id
+                + "/fileServices/default"
+            ),
```
Comment (jemrobinson) [inline data_safe_haven/infrastructure/programs/sre/data.py]:
```suggestion
            # This is the URI of the automatically created fileService resource
            resource_uri=Output.concat(storage_account_data_configuration.id, "/fileServices/default"),
```

File: data_safe_haven/infrastructure/programs/sre/data.py [hunk 3]
```diff
@@ -625,6 +667,47 @@ def __init__(
...
+                        "enabled": False,
+                    },
+                }
+            ],
+            resource_uri=storage_account_data_private_user.id.apply(
+                # This is the URI of the fileServices resource which is automatically
+                # created.
+                lambda resource_id: resource_id
+                + "/fileServices/default"
+            ),
```
Comment (jemrobinson) [inline data_safe_haven/infrastructure/programs/sre/data.py]:
```suggestion
            # This is the URI of the automatically created fileService resource
            resource_uri=Output.concat(storage_account_data_private_user.id, "/fileServices/default"),
```

**Review by jemrobinson — COMMENTED** [review]
Some small questions/suggestions but mostly this looks good.

File: data_safe_haven/infrastructure/programs/declarative_sre.py (hunk 1 above)
Comment (JimMadge) [reply data_safe_haven/infrastructure/programs/declarative_sre.py:214]: This was a duplicate (got added in a merge somehow 😕), it is further up in the file.

File: data_safe_haven/infrastructure/programs/sre/data.py (hunk 2 above)
Comment (JimMadge) [reply data_safe_haven/infrastructure/programs/sre/data.py]: I was certain there was a helper method to do this but couldn't find it! 😆

**Comment by JimMadge** [comment]
> What does this actually log? Is it file creation/deletion etc?

Yes, the precise details seem to depend on the backend (SMB/NFS) but I can see log events for file/directory access, file creation, file deletion.

**Review by craddm — APPROVED** [review]
LGTM
